subprojects/core/src/main/java/org/gradle/process/internal/worker/DefaultWorkerProcessBuilder.java

```diff
@@ -70,6 +71,7 @@
 
     DefaultWorkerProcessBuilder(JavaExecHandleFactory execHandleFactory, MessagingServer server, IdGenerator<Long> idGenerator, ApplicationClassesInSystemClassLoaderWorkerImplementationFactory workerImplementationFactory, OutputEventListener outputEventListener, MemoryManager memoryManager) {
         this.javaCommand = execHandleFactory.newJavaExec();
+        this.javaCommand.setExecutable(Jvm.current().getJavaExecutable());
```

Comment:
If this was not needed before, does that mean we are shipping a change in how `JavaExecHandleFactory` behaves?
I know this is `internal`, so trying to assess if the change there is on purpose.